src/Buttons/TextAndIconButton.js

```diff
@@ -0,0 +1,91 @@
+// @flow
+
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import { StyleSheet, Text, TouchableHighlight, View } from 'react-native'
+import FAIcon from 'react-native-vector-icons/MaterialIcons'
+
+import { styles } from './style'
+
+type Props = {
+  icon: any,
+  style: StyleSheet.Styles,
+  onPress: Function,
+  title: string | Function
+}
+type State = {
+  pressed: boolean
+}
+
+class TextAndIconButton extends Component<Props, State> {
+  static propTypes = {
+    icon: PropTypes.string.isRequired,
+    style: PropTypes.object.isRequired,
+    onPress: PropTypes.func.isRequired,
+    title: PropTypes.oneOfType([PropTypes.string.isRequired, PropTypes.func.isRequired])
+  }
+  UNSAFE_componentWillMount () {
+    this.setState({
+      pressed: false
+    })
+  }
+  _onPressButton = () => {
+    this.props.onPress()
+  }
+  _onShowUnderlay = () => {
+    this.setState({
+      pressed: true
+    })
+  }
+  _onHideUnderlay = () => {
+    this.setState({
+      pressed: false
+    })
+  }
+
+  render () {
+    const {
+      container,
+      centeredContent,
+      inner,
+      textContainer,
+      iconContainer,
+      text,
+      textPressed,
+      icon,
+      iconPressed,
+      iconSize,
+      underlayColor
+    } = this.props.style
+    return (
+      <TouchableHighlight
+        style={container}
+        onPress={this._onPressButton}
+        onShowUnderlay={this._onShowUnderlay}
+        onHideUnderlay={this._onHideUnderlay}
+        underlayColor={underlayColor}
+      >
+        <View style={centeredContent}>
+          <View style={inner}>
+            <View style={textContainer}>
+              {typeof this.props.title === 'string' && (
+                <Text
+                  style={[styles.text, text, this.state.pressed && textPressed]}
+                  ellipsizeMode={'middle'}
+                  numberOfLines={1}
+                >
+                  {this.props.title + ' '}
+                </Text>
+              )}
+              {typeof this.props.title === 'function' &&
+                this.props.title({ textStyles: [styles.text, text, this.state.pressed && textPressed] })}
+            </View>
+            <View style={iconContainer}>{this.props.icon}</View>
+          </View>
+        </View>
+      </TouchableHighlight>
+    )
+  }
+}
+
+export { TextAndIconButton }
```

Comment:
Maybe do `export class TextAndIconButton` up above instead of having exports separated this way.